dirsrvtests/tests/suites/subentries/subentries_test.py

```diff
@@ -0,0 +1,201 @@
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2021 RED Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ----
+
+# Author: Anton Bobrov <abobrov@redhat.com>
+
+import logging
+import pytest
+import os
+import ldap
+from ldap.controls import LDAPControl
+from lib389.utils import *
+from lib389._constants import *
+from lib389.topologies import topology_st as topo
+from lib389.idm.user import UserAccount, UserAccounts
+
+log = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.tier0
+
+"""
+This BooleanControl class is taken from python-ldap,
+see https://www.python-ldap.org/ for details.
+The reason is python-ldap standard class has not been
+updated for Python 3 properly and behaves incorrectly.
+When python-ldap is fixed this class can be removed in
+exchange for from ldap.controls import BooleanControl
+"""
+class BooleanControl(LDAPControl):
+  """
+  Base class for simple request controls with boolean control value.
+  Constructor argument and class attribute:
+  booleanValue
+    Boolean (True/False or 1/0) which is the boolean controlValue.
+  """
+  boolean2ber = { 1:b'\x01\x01\xFF', 0:b'\x01\x01\x00' }
+  ber2boolean = { b'\x01\x01\xFF':1, b'\x01\x01\x00':0 }
+
+  def __init__(self,controlType=None,criticality=False,booleanValue=False):
+    self.controlType = controlType
+    self.criticality = criticality
+    self.booleanValue = booleanValue
+
+  def encodeControlValue(self):
+    return self.boolean2ber[int(self.booleanValue)]
+
+  def decodeControlValue(self,encodedControlValue):
+    self.booleanValue = self.ber2boolean[encodedControlValue]
+
+def test_subentries(topo):
+    """Test LDAP Subentries (RFC 3672)
+
+    :id: 5cdb72eb-d227-49c8-9f7a-89314c717a85
+    :setup: Standalone Instance
+    :steps:
+        1. Add test entries and subentries
+        2. Search for normal entries
+        3. Search for subentries
+        4. Search for normal entries and subentries
+        5. Search with subentries control visibility TRUE
+        6. Search with subentries control visibility FALSE
+    :expectedresults:
+        1. Entries and subentries should be added
+        2. Only normal entries are visible
+        3. Only subentries are visible
+        4. Both normal entries and subentries are visible
+        5. Only subentries are visible
+        6. Only normal entries are visible
+    """
+
+    nentries = 5
+
+    # Add normal entries.
+    user = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    for i in range(0, nentries):
+        user1 = user.create_test_user(uid=i)
+
+    # Add subentries.
+    user = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    for i in range(nentries, nentries * 2):
+        user1 = user.create_test_user(uid=i)
+        user1.add("objectclass", "ldapsubentry")
+
+    # Test normal entries first.
+    search_filter = "(objectclass=inetorgperson)"
```

Comment:
I believe this test can be parametrized on search_filter and request_ctrl. 
We have some examples in [paged results test](https://github.com/389ds/389-ds-base/blob/master/dirsrvtests/tests/suites/paged_results/paged_results_test.py).

Reply:
It sure can be and its gonna be neat and compact but i dont feel its worth a stretch in this unit test. If you feel strongly about it feel free to chip in and redo it however you see fit, kept patch changes flag allowed.

Reply:
That's fine for me, thanks.